### PR TITLE
NT-2025: Fix duplicated comments

### PR DESCRIPTION
**Pull request by Arkariang**
# 📲 What
- We had a bug that was causing posting comments when pull to refresh was triggered
- Some refactor have been done on the `CommentsViewHolderViewModel` 
- Added more test cases

# 🤔 Why
- we had issue identifying when a comment needed to be posted and when just displayed.

# 👀 See

- Here a video posting comments, pulling to refresh after commenting, and also trying error scenarios on the optimistic posting 

https://user-images.githubusercontent.com/4083656/121427807-fcf9bf00-c929-11eb-9d98-583fecf04268.mp4

|  |  |

# 📋 QA
- Add new comments, pull to refresh
- Add new comments, pull to refresh, paginate
- Add new comments, paginate
- try to post without connection, you'll see the error states, makes sure we do not post duplicated comments once you recover the connectivity 

# Story 📖

[NT-2025](https://kickstarter.atlassian.net/browse/NT-2025)

### CODE REVIEW

File: app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt [hunk 1]
```diff
@@ -199,48 +223,72 @@ interface CommentsViewHolderViewModel {
...
             Observable
-                .combineLatest(onRetryViewClicked, commentData) { _, commentData ->
-                    return@combineLatest executePostCommentMutation(commentData)
+                .combineLatest(onRetryViewClicked, commentData) { _, newData ->
+                    return@combineLatest executePostCommentMutation(newData)
                 }
                 .switchMap {
                     it
                 }.doOnNext {
                     this.commentCardStatus.onNext(CommentCardStatus.POSTING_COMMENT_COMPLETED_SUCCESSFULLY)
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt:254]: @hadia  I think this `CommentCardStatus.POSTING_COMMENT_COMPLETED_SUCCESSFULLY` can be removed same as the `doOnNext` but let me know what you think about it

Comment (hadia) [reply app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt:254]: That's when the state to show the green check and Posted keyword

**Review by hadia — COMMENTED** [review]
great 👌🎉

**Review by hadia — APPROVED** [review]
Tested 🎉

**Pull request merged**